types/src/deploy.rs

```diff
@@ -1093,6 +935,191 @@ impl Deploy {
     }
 }
 
+impl hash::Hash for Deploy {
+    fn hash<H: hash::Hasher>(&self, state: &mut H) {
+        // Destructure to make sure we don't accidentally omit fields.
+        #[cfg(any(feature = "once_cell", test))]
+        let Deploy {
+            hash,
+            header,
+            payment,
+            session,
+            approvals,
+            is_valid: _,
```

Comment:
I wonder if one can add this `#[cfg(any(feature = "once_cell", test))]` to the field only, that way it would be destructured only in one place. Not sure if that's desirable, though.

Reply:
I did try that, but it's not valid syntax apparently :)

Reply:
Oops, yeah, the Rust compiler has some quirks, especially when it commes to feature-gated code. I have experienced similar issues in the past (for example, you cannot feature-gate a generic bound).